### PR TITLE
add book's Development section

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: doc/book/src/development/e2e/run-tests.md [hunk 1]
```diff
@@ -1 +1,66 @@
...
+You need to define some variables to use in the next steps.
+
+```sh
+# the tag to use for KubeSaw images build at step 1
+export IMAGE_TAG=e2e-test
+
+# the quay.io namespace to use in the next steps
+export QUAY_NAMESPACE=my-quay-namespace
+```
+
```
Comment (sadlerap) [inline doc/book/src/development/e2e/run-tests.md:23]: Maybe call out the `IMAGE_BUILDER` env var?

File: doc/book/src/development/rest-api.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Following the flow of an HTTP Request, the request will be initially processed by the code in the [`rest` package][server-rest-folder].
+It validates the HTTP Request and builds a Command or Query to use for invoking the handlers in the [`core` package][server-core-folder].
+The `core` package performs validation, authorization, may apply some transformation, before invoking the logic in the [`persistence` package][server-persistence-folder].
+In case of a Command it will perform update or create, in case of a Query it will retrieve some data.
+Finally, the `core` will build a Response and provide it back to the `rest` package which will map it to an HTTP Response.
+
+## Run Tests
+
+To run Unit tests you can execute the `make test` command.
```
Comment (sadlerap) [inline doc/book/src/development/rest-api.md:22]:
```suggestion
To run unit tests you can execute the `make test` command.
```

File: doc/book/src/development/operator.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The [operator code][operator-folder] is using the [Operator-SDK][operator-sdk].
+
+* the [`api` folder][operator-api-folder] contains the Go code for Operator's CRDs
+* the [`config` folder][operator-config-folder] contains the YAML manifests
+* the [`internal` folder][operator-internal-folder] contains the code for the reconcilers 
+
+## Run Tests
+
+To run Unit tests you can execute the `make test` command.
```
Comment (sadlerap) [inline doc/book/src/development/operator.md:11]:
```suggestion
To run unit tests you can execute the `make test` command.
```

Also, it's at this point that I realize we should probably have a top-level `unit-test` rule that runs all unit tests.  IMO there's not any real good reason that unit tests shouldn't run everything.

**Review by dperaza4dustbit — APPROVED** [review]

File: doc/book/src/development/e2e/run-tests.md (hunk 1 above)
Comment (filariow) [reply doc/book/src/development/e2e/run-tests.md:23]: good idea, I'll do this a new PR

Comment (filariow) [reply doc/book/src/development/e2e/run-tests.md:23]: #223

File: doc/book/src/development/rest-api.md (hunk 2 above)
Comment (filariow) [reply doc/book/src/development/rest-api.md:22]: done in #224

File: doc/book/src/development/operator.md (hunk 3 above)
Comment (filariow) [reply doc/book/src/development/operator.md:11]: typo fixed in #224.

I agree on the top level `unit-test` rule. I'll open a PR for this.

Comment (filariow) [reply doc/book/src/development/operator.md:11]: PR for top-level `unit-test` rule: #225

**Pull request merged**